src/v/cluster/partition.cc

```diff
@@ -542,14 +542,6 @@ ss::future<> partition::stop() {
         co_await _archiver->stop();
     }
 
-    if (_archival_meta_stm) {
-        vlog(
-          clusterlog.debug,
-          "Stopping archival_meta_stm on partition: {}",
-          partition_ntp);
-        co_await _archival_meta_stm->stop();
```

Comment:
I can't see where the stm stopping is happening now

Reply:
stm lifecycle is now managed in the `state_machine_manager` i.e. it starts and stops all managed STMs

Reply:
I was curious where it's invoked now (the entry point); already found - from consensus on its stop